### DISCUSSION TITLE
Frigate models - anything else I could try? (False detections)

### DISCUSSION THREAD

**Discussion by LaurenceGough** [blakeblackshear/frigate#5916] [General]
Hello all,

I have read a few posts on this - and I know quite a bit of testing has took place and the model that ships with Frigate was found to be the best balance on the market between inference speed and accuracy.

If say you wanted more accuracy at the expense of inference speed, is there a better model available (perhaps with a higher resolution / input size??). Anyone tried any?

I've tried some CCTV optimized ones, but they sadly performed far worse, I tried the spaghettinet one which seems about on par with the stock one.

I am really struggling with false detections, I have fine tuned min area sizes, confidence level percentages etc, but sadly there is no combination that works without a load of false positives (mainly at night), which I think is fair in saying is one of the main reasons many of us turned to Frigate. I've almost got more person object masks then I have actual cctv feed (ha!).

I know Frigate+ is in the works but just in the meantime...

Thank you

**Comment by NickM-27** [comment]
it seems the default model works much better for some users than others, most likely depending on camera view. 

There are a number of different models which can be tried https://coral.ai/models/object-detection/ which have varying levels of inference speed and precision. It has been found that generally, larger input size does not lead to better accuracy in the frigate usecase

**Reply by LaurenceGough** [reply]
Thanks Nick! I had a look there before but I wasn't sure what to go for, also the "These are not production-quality models; they are for demonstration purposes only." put me off! Your post gave me some confidence to give efficientdet_lite3_512_ptq_edgetpu.tflite a go, seems like the best accuracy vs inference speed, although as you say it might not relate to Frigate at all!

I'll report back, I couldn't find many discussions on this topic, unless I am using the wrong search terms perhaps... 

Now the green bounding box for object detection is much larger, which makes me feel that it's sort of pointless for my detection camera resolution ( 640x480 ) in fact it's larger than the height... ha. Perhaps I should go down a size. I'll see how it goes first and I think I need to have a good read on this object recognition (input size) part again to try to work out more accurately how it works.


No false positives yet, which is good. 330-400ms inference speed is really not so good with a USB Coral but I guess to be expected.  I think the next size or two down would make sense. I'll report back.

**Reply by NickM-27** [reply]
I believe one user anecdotally said that MobileNet V2 worked better for them

**Comment by blakeblackshear** [comment]
Have you tried increasing your detect resolution to 720p or 1080p? At 640x480, a person would need to be about 75% of the frame height to fill a 300x300px region. I would not expect reliable detection for objects any smaller than 50% of the height of the frame at that resolution.

**Reply by LaurenceGough** [reply]
Thanks Blake great to note down. Yes, it sure does seem low, sadly that's as high as the substream can go. I've un-commented out using the main stream (only 1080p on this one), and the related masks and zones that I kept commented out luckily. I cannot recall why I shifted back last time but I'm sure I'll find out why soon.

It's really annoying too as my other camera's max substream resolution is 630x360 but the main stream on that is 2k so I assume that would be too much to use as the detect role too.  Luckily that one hasn't been so bad. What's the downsides of using a high res for the detection input, I guess it could cause more load for CPU motion detection and perhaps a bit tricker for the Coral (or other TPUs) too? 

P.S the other model seems to have fixed the false detection anyway, but inference speeds were increased a fair bit more... (200ms the efficientdet_lite1_384_ptq_edgetpu file I was using).

Thanks

**Reply by NickM-27** [reply]
The coral is not affected by higher resolution as it only ever takes in one size image. The CPU is used to downsize the region for detection before sending to the coral.

If you had a more capable device like intel, then you could use the GPU to scale down the 2K stream to 1080p creating a balance between CPU usage and accuracy. Unfortunately, the RPi does not have hw accelerated scaling so it would run on the CPU which would likely bog it down quite a bit.

**Reply by LaurenceGough** [reply]
Thank you Nick, that makes sense. Yeah, I'm looking out for a suitable Intel powered mini PC :) that would be nice to allow for some more options such as that.

**Comment by aaronjrodrigues** [comment]
Hey man, in the same boat as you. Wanted to ask what model you decided on.

**Reply by LaurenceGough** [reply]
Hey, I'm now using openvino with /openvino-model/ssdlite_mobilenet_v2.xml and the model configuration as in the setup instructions.

It's like a whole new system now and there are hardly any false positives, but that could also be down to the other Frigate 14 improvements perhaps?

black and white IR detection is still being difficult sometimes, but I could probably do with IR flood lights... I mean I can clearly see on the stream but it still struggles with detecting people.

**Reply by Plastikman81** [reply]
Does the openvino still use the Coral or does it uses the igpu?
Could you recommend a Guide or Video how to configure? I didnt finde any Guide how to Setup.
Thanks in advance...
Regards

**Reply by hawkeye217** [reply]
OpenVINO uses your Intel GPU. The official documentation shows how to configure it: https://docs.frigate.video/configuration/object_detectors#openvino-detector

**Comment by edurenye** [comment]
Is it possible to have a fast model that has false detections and then send it to a better but slower model to filter them?
